src/test/scala/io/smartdatalake/testutils/DataFrameTestHelper.scala

```diff
@@ -0,0 +1,274 @@
+        case (name: String, _: DecimalType) => (intPrefix(name), IntegerType)
+        case (name: String, dataType: Any) => (name, dataType)
+      }
+
+    // Create spark schema out of column types
+    val structFields: Seq[StructField] =
+      namesAndTypeWithTimestampAsString.map {
+        case (name: String, dataType: DataType) => StructField(name, dataType, nullable = true)
+      }
+    val schema: StructType = StructType(structFields)
+
+    // create a table containing the value or else the default value or else null for each row and column
+    // creates a list of rows, where a row is a list of values of Any type
+    val rowValues: Seq[Seq[Any]] =
+    input.map(inputRow =>
+      columnNames.map(columnName =>
+        inputRow.get(columnName) match {
+          case Some(valueAndType) => valueAndType.value
+          case None => defaultValues.get(columnName) match {
+            case Some(valueAndType) => valueAndType.value
+            case None => null
+          }
+        }
+      )
+    )
+
+    // Convert rows to spark rows
+    val rowData: Seq[Row] =
+      rowValues.map(row => Row(row: _*))
+
+    // Initialize test spark session and create DataFrame with the values and schema
+    val spark = SparkSession.builder().master("local").getOrCreate()
+    val rdd: RDD[Row] = spark.sparkContext.parallelize(rowData)
+    val df = spark.createDataFrame(rdd, schema)
+
+    // Special handling of Timestamp and Decimal columns: Steps 2 and 3
+    val castTimestampColumn: (DataFrame, String) => DataFrame = (df, columnName) =>
+      df.withColumn(columnName, to_timestamp(col(stringPrefix(columnName))))
+        .drop(stringPrefix(columnName))
+    val castDecimalColumn: (DataFrame, String) => DataFrame = (df, columnName) =>
+      df.withColumn(columnName, col(intPrefix(columnName)).cast(DecimalType(38, 0)))
+        .drop(intPrefix(columnName))
+
+    val dfTransformations = namesAndTypes.map {
+      case (columnName: String, _: TimestampType) => (df: DataFrame) => castTimestampColumn(df, columnName)
+      case (columnName: String, _: DecimalType) => (df: DataFrame) => castDecimalColumn(df, columnName)
+      case (_: String, _: Any) => (df: DataFrame) => df
+    }
+
+    val dfResult = Function.chain(dfTransformations)(df)
+
+    // done!
+    dfResult
+  }
+
+  def createDf(input: Map[String, TypedValue]*): DataFrame = {
+    createDfWithDefaultValues(input: _*)()
+  }
+
+  implicit def valueToTypedValue[T](value: T): TypedValue = value match {
+    case (value: Any, dataType: DataType) => TypedValue(value, dataType)
+    case (null, dataType: DataType) => TypedValue(null, dataType)
+    case string: String => TypedValue(string, StringType)
+    case int: Int => TypedValue(int, IntegerType)
+    case bool: Boolean => TypedValue(bool, BooleanType)
+    case _ => throw new Exception("Unable to convert to TypedValue")
+  }
+
+  def assertSchemasEqual(expected: StructType, actual: StructType): Unit = {
```

Comment:
In assertDataFramesEqual we have another implementation of schema comparision. Could we merge them?
I would prefere the one from assertDataFramesEqual.